libraries/chain/worker_evaluators.cpp

```diff
@@ -206,9 +206,82 @@ namespace golos { namespace chain {
                 _db.modify(*wpo_itr, [&](worker_proposal_object& wpo) {
                     wpo.approved_techspec_author = o.author;
                     from_string(wpo.approved_techspec_permlink, o.permlink);
+                    wpo.state = worker_proposal_state::techspec;
                 });
             }
         }
     }
 
+    void worker_result_fill_evaluator::do_apply(const worker_result_fill_operation& o) {
+        ASSERT_REQ_HF(STEEMIT_HARDFORK_0_20__1013, "worker_result_fill_operation");
+
+        const auto now = _db.head_block_time();
+
+        GOLOS_CHECK_LOGIC(o.completion_date <= now,
+            logic_exception::work_completion_date_cannot_be_in_future,
+            "Work completion date cannot be in future");
+
+        const auto& comment = _db.get_comment(o.author, o.permlink);
+
+        GOLOS_CHECK_LOGIC(comment.parent_author == STEEMIT_ROOT_POST_PARENT,
+            logic_exception::worker_result_can_be_created_only_on_post,
+            "Worker result can be created only on post");
+
+        const auto& wto_idx = _db.get_index<worker_techspec_index, by_permlink>();
+        auto wto_itr = wto_idx.find(std::make_tuple(o.author, o.worker_techspec_permlink));
+
+        if (wto_itr == wto_idx.end()) {
+            GOLOS_THROW_MISSING_OBJECT("worker_techspec_object", fc::mutable_variant_object()("author",o.author)("permlink",o.worker_techspec_permlink));
+        }
```

Comment:
Why we can't use `get_worker_techspec()` in this place?

Reply:
@afalaleev I added `find_worker_result` because it was possible. But all another functions aren't available now. However they will be available when I'll resolve conflicts. It is best to do it when resolving conflicts.